com.dynamo.cr/com.dynamo.cr.bob/src/com/defold/extension/pipeline/ILuaTranspiler.java

```diff
@@ -0,0 +1,114 @@
+// Copyright 2020-2024 The Defold Foundation
+// Copyright 2014-2020 King
+// Copyright 2009-2014 Ragnar Svensson, Christian Murray
+// Licensed under the Defold License version 1.0 (the "License"); you may not use
+// this file except in compliance with the License.
+//
+// You may obtain a copy of the License, together with FAQs at
+// https://www.defold.com/license
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+package com.defold.extension.pipeline;
+
+import java.io.File;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Implement this interface in com.defold.extension.pipeline package with no-argument constructor
+ * to add support for transpiling a language to Lua that is then will be used during the build
+ */
+public interface ILuaTranspiler {
+
+    /**
+     * @return A resource path to a build file, a project-relative path starting with slash,
+     * e.g. {@code "/tlconfig.lua"}
+     */
+    String getBuildFileResourcePath();
+
+    /**
+     * @return A file extension for the source code files of the transpiled language,
+     * e.g. {@code "ext"}
+     */
+    String getSourceExt();
```

Comment:
Would it be beneficial to be able to return multiple file extensions that a `ILuaTranspiler` could handle?

Reply:
Good question. Is it very likely to be a thing though? We could always change it later if it comes to that. Although maybe it is better to do it now and avoid having to break things down the line?

Reply:
When I decided to use a single extension, I considered if we will need to introduce a breaking change down the line. First of all, I looked at many languages that transpile to lua (here is a big list of such languages — https://github.com/hengestone/lua-languages), and all of them only used a single extension for source files. Even for semantically different types of sources, languages still use the same extension (for example, for external definition files, teal and typescript use `.d.tl` and `.d.ts` extensions, which as still essentially `.tl` and `.ts`). In case where we indeed will need to support multiple source extensions, we can add support for them without breaking by introducing a default interface method:
```java
    default List<String> getSourceExts() {
        return Collections.singletonList(getSourceExt());
    }
```
This way, existing implementations may stay the same, and new implementations may support multiple source code extensions. I don't think we need this yet because it's still purely hypothetical.

Reply:
I don't think the default interface method solution solves the issue completely. Wouldn't that mean that implementors that require more than one source ext would need to implement both methods, even though only one of them have an implementation that makes sense?

Reply:
I was thinking about also changing `getSourceExt` to default that throws an error:
```java
    default String getSourceExt() {
        throw new IllegalStateException("implement either getSourceExt() or getSourceExts()");
    }
```
This way, implementors will only need to override one of the methods.

Reply:
I guess that's valid.